src/tests.rs

```diff
@@ -13,6 +17,35 @@ macro_rules! create_tests {
     ($($name:ident: $value:expr,)*) => { $(
         #[tokio::test(flavor = "multi_thread")]
         async fn $name() -> Result<ExitCode> {
+            // For the formatTime test, we need to enable the fr_FR locale for UTF-8 for the test to pass
+            if stringify!($name) == "datetime_format_time" {
+                // Create a new scope, so the file gets closed after it is out of scope
+                {
+                    // Creates a bash script for appending to /etc/locales.gen
+
+                    // Using a bash script simplifies a lot of piping logic that would
+                    // otherwise be required in Rust
+                    let mut append_locale_script = File::options().write(true).create(true).mode(0o110).open("target/datetime_format_time_locales.sh").await?;
+                    append_locale_script.write_all(b"echo fr_FR.UTF-8 UTF-8 | tee -a /etc/locale.gen").await?;
+                };
+
+                // Inherits the credentials from parent stdin
+                let out = Command::new("sudo").arg("-S").arg("bash").arg("-c").arg("target/datetime_format_time_locales.sh").stdin(Stdio::inherit()).output().await?;
+
+                if !out.status.success() {
+                    eprintln!("ERROR: Failed to write locale info to /etc/locale.gen, could not run as root");
+                    eprintln!("{}", String::from_utf8_lossy(&out.stderr));
+                    return Ok(ExitCode::FAILURE);
+                }
+
+                let locale_gen = Command::new("sudo").arg("-S").arg("locale-gen").stdin(Stdio::inherit()).output().await?;
+
+                if !locale_gen.status.success() {
+                    eprintln!("ERROR: Failed to run locale-gen as root");
+                    eprintln!("{}", String::from_utf8_lossy(&out.stderr));
+                    return Ok(ExitCode::FAILURE);
+                }
+            }
```

Comment:
I dont think we should modify the test runner itself in test-specific or platform-specific ways like this. People are going to be running these tests on many platforms locally, us not doing it in CI as well (which would make this fail) is an oversight on my part, so I'll get that sorted out. It's fine to not test the locale-specific part of `formatTime` imo since there's really no great testing solution here

Reply:
You're right, considering that chrono-locale already has tests on its side.

Reply:
I think the best path forward if we still want to keep that later part of the test would be to put it behind some user-provided flag in `process.args`, and for anyone making changes to DateTime locales to run these tests with that flag locally.
`chrono_lc` already has a much more rigorous test suite and all we really need to do is verify that the locale mechanism is being used *at all*, and not its behavior, which really is fine to do as a one-off local test run thing.

```lua
-- formatTime.luau

local process = require("@lune/process")

-- ... normal formatting tests ...

local runLocaleTest = false
for _, arg in process.args do
    if arg == "--test-locales" then
        runLocaleTest = true
        break
    end
end
if not runLocaleTest then
    return
end

-- ... locale-specific tests ...
```

```bash
cargo run -- tests/datetime/formatTime --test-locales
```

Note that this doesn't actually work atm, `process.args` is completely static for tests, but I'll get that sorted out, too

Reply:
Fixed in https://github.com/filiptibell/lune/pull/94/commits/fcb20571158911188c0e651e84f4155e5969d448.